### PR TITLE
fix Typescript error TS2349 in .d.ts file

### PR DESCRIPTION
**Pull request by mattbishop**
Version 1.2, when used in a Typescript project, emits the following error:

```error TS2349: Cannot invoke an expression whose type lacks a call signature. Type 'typeof import("/Users/mbishop/OCTO/atlantis/node_modules/deep-freeze-es6/index")' has no compatible call signatures.```

This fix uses a TS declaration file created by dts-gen.

### CODE REVIEW

**Comment by mattbishop** [comment]
Not sure what you are saying, commitlint. I have a commit message. Not sure what "type" means here. Please instruct.

**Comment by christophehurpeau** [comment]
Thanks for your PR, I will look into that before the end of the week.
I follow http://conventionalcommits.org/, but don't worry about it I can change the commit name when I "Squash and merge"

**Comment by mattbishop** [comment]
Cool. I thought about creating a unit test that loads TSC and compiles the test file, but that seemed like a lot of weight when there aren't any tests now. I tested this by loading the project in VS Code and seeing if the test file had any errors.

File: index.d.ts [hunk 1]
```diff
@@ -1,2 +1,9 @@
+/** Declaration file generated by dts-gen */
+
+export = deepFreeze;
+
 declare function deepFreeze<T>(obj: T): T;
-export default deepFreeze;
+
+declare namespace deepFreeze {
+    const prototype: {};
```
Comment (christophehurpeau) [inline index.d.ts]: there is no prototype anywhere in the lib

File: test-deep-freeze.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import * as deepFreeze from "./";
```
Comment (christophehurpeau) [inline test-deep-freeze.ts]: `deepFreeze` should not be imported as *, there is only one default export.

**Comment by mattbishop** [comment]
Christophe, I removed the `prototype` declaration as you suggest and thankfully the TS test was fine with the change. I've pushed this to the PR.

About the import statement, you are correct in that there is a single function, but this statement is more of an aliasing mechanism than an 'import everything' mechanism. One could easily write `import * as banana` and use `banana(mydata)`.

**Comment by christophehurpeau** [comment]
@mattbishop can you look at #5 ?

**Pull request closed without merging**